### PR TITLE
Group analyses

### PR DESCRIPTION
**Pull request by dchiquit**
There is now an Analysis selector where you can choose one of:
* All Metabolites
* Selected Metabolites
* Factor Analysis

The Metabolite Source selector now only appears when Factor Analysis is
selected, and will only contain the PCs.

Also, when `usePlotData` was converted from the `plotData` mixin, I missed a spot
where it invalidated the plot whenever the arguments changed. This means
that new TS plots were not triggering refreshes whenever they should
have. That is now fixed in `usePlotData`.

### CODE REVIEW

**Review by subdavis — CHANGES_REQUESTED** [review]
I think better messaging to the user is needed here.  It's not clear which fields are required to make the vis load.

* I expected "All metabolites" to use all the metabolites, not populate the second list with all the metabolites as options.   
* Selected metabolites is also ambiguous, since there is a selection box below.  I originally thought it would be the selections from the next box, but it's the global selections
* In cases where I have no global selections, nothing tells me this is a problem: it just doesn't load.

IMO, all 5 form elements need some visual validation indicator to tell me which ones are required and what I did wrong.  This all makes sense now that I know how it works, and I understand why it works like this.

## Selection is overloaded.

I should capture this is another issue for later discussion.  

In general, the UX around our global selected set is pretty confusing.  One problem is that the term "selection" is way too overloaded in this app.  We probably need a unique phrase to identify this ability to globally choose a group of metabolites across analyses.

This could even tie into our marketing: we could use this phrase to identify the feature as a particular strength of the tool.  Something like "Interest group".  Or we could identify the repository almost like a shopping cart: "Add this metabolite to your magnifying glass?" These are objectively terrible names, but you get the point.  If we worked for some terrible enterprise software company, this is the sort of thing the marketing team would call "TrueSelect Technology®"

Tagging @waxlamp because I think he'll probably have input.

Screenshot from 2020-11-03 08-49-21

**Comment by waxlamp** [comment]
> I think better messaging to the user is needed here. It's not clear which fields are required to make the vis load.
> 
> * I expected "All metabolites" to use all the metabolites, not populate the second list with all the metabolites as options.
> * Selected metabolites is also ambiguous, since there is a selection box below.  I originally thought it would be the selections from the next box, but it's the global selections
> * In cases where I have no global selections, nothing tells me this is a problem: it just doesn't load.
> 
> IMO, all 5 form elements need some visual validation indicator to tell me which ones are required and what I did wrong. This all makes sense now that I know how it works, and I understand why it works like this.
> 
> ## Selection is overloaded.
> I should capture this is another issue for later discussion.
> 
> In general, the UX around our global selected set is pretty confusing. One problem is that the term "selection" is way too overloaded in this app. We probably need a unique phrase to identify this ability to globally choose a group of metabolites across analyses.
> 
> This could even tie into our marketing: we could use this phrase to identify the feature as a particular strength of the tool. Something like "Interest group". Or we could identify the repository almost like a shopping cart: "Add this metabolite to your magnifying glass?" These are objectively terrible names, but you get the point. If we worked for some terrible enterprise software company, this is the sort of thing the marketing team would call "TrueSelect Technology®"
> 
> Tagging @waxlamp because I think he'll probably have input.
> 
> Screenshot from 2020-11-03 08-49-21

You should definitely transplant this comment into a new issue. I'd do it myself but I want you to be shown as the author of it.

Once you do that, I'll respond with my thoughts there.

**Comment by subdavis** [comment]
Added #692

**Comment by dchiquit** [comment]
Outside of the selection discussion, I'm going to save the group prediction usability discussion for later as well. This is a preliminary PR for adding the PLSDA/OPLSDA analyses to the group prediction page, so I think it will be helpful to have all the functionality in place before tweaking for usability. I don't foresee any drastic overhauls, just some rearrangement and adding tooltips/expository text.

File: web/src/components/vis/GroupPredictionTile.vue [hunk 1]
```diff
@@ -1,144 +1,197 @@
...
-        this.pcaData = null;
+        pcaData.value = null;
       }
-    },
+    }
+
+    // Perform an initial factor analysis when component is mounted
+    onMounted(() => {
+      getFactors();
+    });
```
[the remark this answers is not in the document]
Comment (dchiquit) [reply web/src/components/vis/GroupPredictionTile.vue]: I just copied this straight from the original implementation. I guess in the Vue 2 world you need to add initialization stuff in some kind of hook, so Mike (I think he implemented this?) chose mounted. Now it makes more sense to just trigger it in the setup function.

**Review by subdavis — APPROVED** [review]

**Pull request merged**